File: system-test/java/src/test/java/com/mycompany/myshop/systemtest/latest/acceptance/CancelOrderNegativeTest.java
package com.mycompany.myshop.systemtest.latest.acceptance;

import com.mycompany.myshop.systemtest.latest.acceptance.base.BaseAcceptanceTest;
import com.mycompany.myshop.testkit.channel.ChannelType;
import com.mycompany.myshop.testkit.domainvaluetypes.OrderStatus;
import com.optivem.testing.Channel;
import com.optivem.testing.DataSource;
import org.junit.jupiter.api.TestTemplate;

class CancelOrderNegativeTest extends BaseAcceptanceTest {
    @TestTemplate
    @Channel({ChannelType.API})
    @DataSource({"NON-EXISTENT-ORDER-99999", "Order NON-EXISTENT-ORDER-99999 does not exist."})
    @DataSource({"NON-EXISTENT-ORDER-88888", "Order NON-EXISTENT-ORDER-88888 does not exist."})
    @DataSource({"NON-EXISTENT-ORDER-77777", "Order NON-EXISTENT-ORDER-77777 does not exist."})
    void shouldNotCancelNonExistentOrder(String orderNumber, String expectedErrorMessage) {
        scenario
                .when().cancelOrder()
                    .withOrderNumber(orderNumber)
                .then().shouldFail()
                    .errorMessage(expectedErrorMessage);
    }

    @TestTemplate
    @Channel({ChannelType.API})
    void shouldNotCancelAlreadyCancelledOrder() {
        scenario
                .given().order()
                    .withStatus(OrderStatus.CANCELLED)
                .when().cancelOrder()
                .then().shouldFail()
                    .errorMessage("Order has already been cancelled");
    }

    @TestTemplate
    @Channel({ChannelType.API})
    void cannotCancelNonExistentOrder() {
        scenario
                .when().cancelOrder()
                    .withOrderNumber("non-existent-order-12345")
                .then().shouldFail()
                    .errorMessage("Order non-existent-order-12345 does not exist.");
    }
}

File: system-test/dotnet/SystemTests/Latest/AcceptanceTests/CancelOrderNegativeTest.cs
using SystemTests.Latest.AcceptanceTests.Base;
using Dsl.Core.UseCase;
using Driver.Port.Dtos;
using DomainValueTypes;
using Optivem.Testing;

namespace SystemTests.Latest.AcceptanceTests;

public class CancelOrderNegativeTest : BaseAcceptanceTest
{
    [Theory]
    [ChannelData(ChannelType.API)]
    [ChannelInlineData("NON-EXISTENT-ORDER-99999", "Order NON-EXISTENT-ORDER-99999 does not exist.")]
    [ChannelInlineData("NON-EXISTENT-ORDER-88888", "Order NON-EXISTENT-ORDER-88888 does not exist.")]
    [ChannelInlineData("NON-EXISTENT-ORDER-77777", "Order NON-EXISTENT-ORDER-77777 does not exist.")]
    public async Task ShouldNotCancelNonExistentOrder(Channel channel, string orderNumber, string expectedErrorMessage)
    {
        await Scenario(channel)
            .When().CancelOrder().WithOrderNumber(orderNumber)
            .Then().ShouldFail()
            .ErrorMessage(expectedErrorMessage);
    }

    [Theory]
    [ChannelData(ChannelType.API)]
    public async Task ShouldNotCancelAlreadyCancelledOrder(Channel channel)
    {
        await Scenario(channel)
            .Given().Order().WithStatus(OrderStatus.Cancelled)
            .When().CancelOrder()
            .Then().ShouldFail()
            .ErrorMessage("Order has already been cancelled");
    }

    [Theory]
    [ChannelData(ChannelType.API)]
    public async Task CannotCancelNonExistentOrder(Channel channel)
    {
        await Scenario(channel)
            .When().CancelOrder().WithOrderNumber("non-existent-order-12345")
            .Then().ShouldFail()
            .ErrorMessage("Order non-existent-order-12345 does not exist.");
    }
}
